azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -1372,3 +1411,63 @@ func resourceArmContainerGroupPortsHash(v interface{}) int {
 
 	return hashcode.String(buf.String())
 }
+
+func flattenContainerGroupDnsConfig(input *containerinstance.DNSConfiguration) []interface{} {
+	output := make(map[string]interface{})
+	var outputArr []interface{}
+
+	if input == nil {
+		return outputArr
+	}
+
+	//We're converting to TypeSet here from an API response that looks like "a b c" (assumes space delimited)
+	var searchDomains []string
+	if input.SearchDomains != nil {
+		searchDomains = strings.Split(*input.SearchDomains, " ")
+	}
+	output["search_domains"] = searchDomains
+
+	//We're converting to TypeSet here from an API response that looks like "a b c" (assumes space delimited)
+	var options []string
+	if input.Options != nil {
+		options = strings.Split(*input.Options, " ")
+	}
+	output["options"] = options
+
+	//Nameservers is already an array from the API
+	var nameservers []string
+	if input.NameServers != nil {
+		nameservers = *input.NameServers
+	}
+	output["nameservers"] = nameservers
+
+	outputArr = append(outputArr, output)
+	return outputArr
+}
+
+func expandContainerGroupDnsConfig(d *schema.ResourceData) *containerinstance.DNSConfiguration {
```

Comment:
Can we change the function signature a bit to accept a `[]interface{}` instead, and modify the first `if` logic a bit as [this example expand function](https://github.com/terraform-providers/terraform-provider-azurerm/blob/aad52564c4fad1197ed4ab03f62e35d0ffc0f63b/azurerm/internal/services/containers/container_group_resource.go#L876)?

(since the type of `dns_config` is always guaranteed to be `schema.Set` in this case)